### PR TITLE
Feature/logic app

### PR DESCRIPTION
**Pull request by mytev**

### CODE REVIEW

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Logic/New-ArmLogicApp.ps1 [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        [string]
+        $Name,
+        [PSTypeName("WorkflowDefinition")]
+        $WorkflowDefinition = @{},
+        [string]
+        $ApiVersion = '2017-07-01',
+        [string]
+        $Location = $script:Location,
+        [string]
+        $State = 'Enabled'
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Logic/New-ArmLogicApp.ps1:17]: Change to a switch named "Disable" and map it to the correct string in this cmdlet

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Logic/New-ArmWorkflowDefinition.ps1 [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        [switch]
+        $WithouHttpTrigger,
+        [PSCustomObject]
+        $HttpTriggerSchema = @{}
+    )
+
+    If ($PSCmdlet.ShouldProcess("Creates a new Workflow Definition")) {
+        $Triggers = @{}
+
+        If ($WithouHttpTrigger.ToBool() -ne $True) {
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Logic/New-ArmWorkflowDefinition.ps1]: could be replace by `-not`

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Logic/Add-ArmWorkflowHttpAction.ps1 [hunk 3]
```diff
@@ -0,0 +1,34 @@
+function Add-ArmWorkflowHttpAction {
```
Comment (mbernard) [inline src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Logic/Add-ArmWorkflowHttpAction.ps1]: I don't think we want to create a cmdlet for each possible action. I suggest we make it a way more generic. WorkflowDefinition should be considered as data and not structure. I suggest we take in a json object as the input of the New-ArmLogicApp function

**Review by mbernard — CHANGES_REQUESTED** [review]

File: src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Logic/New-ArmLogicApp.ps1 (hunk 1 above)
Comment (mytev) [reply src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Logic/New-ArmLogicApp.ps1:17]: I think there is more than 2 states ("NotSpecified", "Completed", "Enabled", "Disabled", "Deleted", "Suspended")
Do we only add support for Enabled and Disabled?

Comment (mbernard) [reply src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Logic/New-ArmLogicApp.ps1:17]: haaaa perfect then

**Review by mbernard — APPROVED** [review]

**Pull request merged**